Java/cp/MaximumEachSubarray.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Maximum of All Subarrays of Size k
+ * Java program to find the maximum integer in each subarray of 
+ * fixed size k, given an array of integers, using the 'Sliding 
+ * Window Technique'. 
+ */
+
+import java.io.*;
+import java.util.*; 
+public class MaximumEachSubarray {
+
+    private static String findMaxEachSubarray(int[] numArray, int k) {
+        String result = ""; 
+
+        //ArrayDeque stores indices of useful elements for each window.
+        //These indices are sorted in decreasing order, so that for each
+        //window, index of maximum element is at the front. 
+        ArrayDeque<Integer> indices = new ArrayDeque<>();
+        
+        //For first window 
+        for(int i=0; i<k; i++){
+
+            //For any element, all elements smaller than it and to its left in 
+            //the window are useless as they cannot be the maximum for that 
+            //window. Hence, we remove them from rear of deque. 
+            while(!indices.isEmpty() && (numArray[i] >= numArray[indices.peekLast()]))
+                indices.removeLast(); 
+
+            //Adding newest element to rear of deque
+            indices.addLast(i); 
+        }
+        
+        //For first window
+        result += (numArray[indices.peek()] + " "); 
+        
+        //The same process is repeated over all windows of length k.
+        for(int i=k; i<numArray.length; i++) {
+
+            //If window has crossed the index at front of deque
+            if(indices.peek() <= (i-k)) 
+                indices.removeFirst(); 
+
+            while(!indices.isEmpty() && (numArray[indices.peekLast()] <= numArray[i]))
+                indices.removeLast(); 
+            indices.addLast(i); 
+            result += (numArray[indices.peek()] + " "); 
+        }
+
+        return result;
+    }
+
+    public static void main(String[] args) throws IOException {
+        InputStreamReader read = new InputStreamReader(System.in);
+	    BufferedReader buf = new BufferedReader(read); 
+
+        //Taking input from user
+        System.out.println("Enter length of array of integers: ");
+	    int len = Integer.parseInt(buf.readLine());
+        System.out.println("Enter array of integers: ");
+	    StringTokenizer st = new StringTokenizer(buf.readLine());
+	    int numArray[] = new int[len];
+	    for(int i=0; i<len; i++) {
+	        numArray[i] = Integer.parseInt(st.nextToken()); 
+	    }
+        System.out.println("Enter size of window: ");
+	    int K = Integer.parseInt(buf.readLine()); 
+	    
```

Comment:
Display a small message what you are printing as output for better user-interface

Reply:
@siddharth25pandey Okay :+1:

Reply:
@neha-b2001 Please read carefully, In form of output not as a comment, same follows for your rest PR

Reply:
@siddharth25pandey Sorry, I'll fix this right away.